Library/Homebrew/tap.rb

```diff
@@ -753,34 +753,34 @@ def cask_files_by_name
     end
   end
 
-  # Check whether the file has a Ruby extension.
-  sig { params(file: Pathname).returns(T::Boolean) }
-  def ruby_file?(file)
-    file.extname == ".rb"
+  sig { returns(Regexp) }
+  def formula_file_regex
+    @formula_file_regex ||= case formula_dir
+    when path/"Formula"
+      %r{^Formula(/[^/]+)+\.rb$}
+    when path/"HomebrewFormula"
+      %r{^HomebrewFormula(/[^/]+)+\.rb$}
+    when path
+      %r{^[^/]+\.rb$}
```

Comment:
```suggestion
    formula_file_regex = "[^/]+\.rb
    file_file_and_dir_regex = "(/[^/])*#{formula_file_regex}"
    when path/"Formula"
      %r{^Formula#{file_file_and_dir_regex}$}
    when path/"HomebrewFormula"
      %r{^HomebrewFormula#{file_file_and_dir_regex}$}
    when path
      %r{^#{formula_file_regex}$}
```
which possibly doesn't work at all but something like this that DRYs up the regex a bit and explains what's going on by breaking into components

Reply:
I'm not a big fan of interpolating regexes based on smaller regexes especially when these regexes are already quite small because it ends up making things harder to read. Maybe adding a comment explaining each regex would be more helpful.

Reply:
I find the regexes as is pretty hard to follow. I don't think everything should always be DRY but, in this case, it is important that these three regexes have the same trailing end and only really differ by prefix. If you can find another way of achieving the same thing with better performance, cool, but, if not: I'd really like to split this into variables so that these don't run the risk of getting out of sync over time.

Reply:
Tests have already been added to prevent this from happening so I don't think we need to worry about things getting out of sync here. IMO the biggest issue here is readability. Maybe something like this would be easier to read. It splits out the directory part of the regex into a separate section from the file name part and adds some inline comments. I haven't tested it yet but it should be close enough.

```rb
%r{
  ^Formula/
  (?:[^/]+/)* # zero or more subdirectories
  [^/]+\.rb   # ruby file name
}x
```

Reply:
> Tests have already been added to prevent this from happening

Tests don't prevent these getting out of sync, they just prevent them from getting out of sync in a way that breaks those specific tests.

> IMO the biggest issue here is readability.

I somewhat agree. I think the biggest issue is that you have three non-variable-assigned regexes and ⅔ of them differ only by prefix and they all have the same suffix. This harms both readability (it's hard to mentally parse where they are the same/not) but also: this is literally shared (and crucial!) logic that _must_ remain consistent (or things will go 💥). 

This is an almost textbook case of when you want to make things DRY. There's no world in which you want them to go out of sync.

> Maybe something like this would be easier to read. 

Do you propose to do these same comments for each of the regexes? This really seems like what variables are for...

> It splits out the directory part of the regex into a separate section from the file name part

Or: you could do this with variables? Please?

Reply:
It sounds like you feel very strongly about this point so I'll update it. In the future, if you feel very strongly about something to the point that it's a blocker, could you use `Changed Requested` to better communicate that?

Reply:
Addressed in https://github.com/Homebrew/brew/pull/18167/commits/60b8878f2176d7698bf789c5fdbbefc821b8ab05

Reply:
I would just like more clarity when it comes to which changes are nits/ignorable, suggestions/optional and blocking/required. It doesn't necessarily have to be expressed as `Changes Requested` but that's the most straightforward way to express that something is blocking/required.